src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -852,17 +852,21 @@ def add_permission(client, identifier, api, api_permissions):
     #         id            <- api_permissions
     #         type
 
-    application = show_application(client, identifier)
-
     resource_access_list = []
     for e in api_permissions:
-        access_id, access_type = e.split('=')
+        try:
+            access_id, access_type = e.split('=')
+        except ValueError as ex:
+            from azure.cli.core.azclierror import ArgumentUsageError
+            raise ArgumentUsageError('Usage error: Permission type should be provided, such as '
+                                     '`--api-permissions e1fe6dd8-ba31-4d61-89e7-88639da4683d=Scope`') from ex
```

Comment:
A small suggestion,  we usually put parsing logic in argument action. For example:
https://github.com/Azure/azure-cli/blob/e1864b0718a626db7a6aa411f431b51077b85cc2/src/azure-cli/azure/cli/command_modules/network/_actions.py#L16-L41

Reply:
I can't say it is a good idea to inherent from protected class `argparse._AppendAction`, given `argparse` is not that stable (causing problems such as https://github.com/Azure/azure-cli/issues/20269).

Also, the 3rd parameter of `super(AddBackendAddressCreate, self).__call__` is `values`:

```
    def __call__(self, parser, namespace, values, option_string=None):
        items = getattr(namespace, self.dest, None)
        items = _copy_items(items)
        items.append(values)
        setattr(namespace, self.dest, items)
```

Why is it renamed to `action`?

Anyway, these lines have been there [since 2018](https://github.com/Azure/azure-cli/blame/710c821b2920df504ae037e7b05763193b051a30/src/azure-cli/azure/cli/command_modules/role/custom.py#L859). Let's don't change them too much to avoid unexpected errors. 😉